src/dsl.rs

```diff
@@ -53,14 +53,29 @@ impl<F, TraceArgs> CircuitContext<F, TraceArgs> {
     }
 
     /// Exposes the first step instance value of a forward signal as public.
-    pub fn expose(&mut self, forward: Queriable<F>) {
-        match forward {
+    pub fn expose(&mut self, queriable: Queriable<F>, offset: ExposeOffset) {
+        match queriable {
             Queriable::Forward(forward_signal, _) => {
-                self.sc.expose(forward_signal);
+                let signal = Signal::Forward(forward_signal);
```

Comment:
@lancenonce I think it is better to pass Queriable to ast::expose, so we don't need to add an enum just for the expose feature